cmd/cl-controlplane/app/server.go

```diff
@@ -230,10 +230,19 @@ func (o *Options) Run() error {
 	readinessListenAddress := fmt.Sprintf("0.0.0.0:%d", api.ReadinessListenPort)
 	httpServer := utilhttp.NewServer("controlplane-http", nil)
 	httpServer.Router().Get("/", func(w http.ResponseWriter, r *http.Request) {
-		resp, err := http.Get(fmt.Sprintf("https://127.0.0.1:%d", api.ListenPort))
+		ctx := r.Context()
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("https://127.0.0.1:%d", api.ListenPort), http.NoBody)
+		if err != nil {
+			logrus.Errorf("Error creating request: %v", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		resp, err := http.DefaultClient.Do(req)
```

Comment:
nit: I would prefer not to use default client (e.g., it has no timeouts), but this can be done separately in a different PR